src/storage/src/controller.rs

```diff
@@ -1962,226 +1962,243 @@ mod persist_write_handles {
-                    let mut all_responses = Vec::default();
-
-                    let mut shutdown = false;
-
-                    while let Some((span, command)) = commands.pop_front() {
-                        match command {
-                            PersistWorkerCmd::Register(id, write_handle) => {
-                                let previous = write_handles.insert(id, write_handle);
-                                if previous.is_some() {
-                                    panic!(
-                                        "already registered a ReadHandle for collection {:?}",
-                                        id
-                                    );
-                                }
+                let mut interval = tokio::time::interval(std::time::Duration::from_secs(60));
+                let mut shutdown = false;
+                while !shutdown {
+                    tokio::select! {
+                        _ = interval.tick() => {
+                            let futs = FuturesUnordered::new();
+                            for (_id, write) in write_handles.iter_mut() {
+                                futs.push(write.maybe_heartbeat_writer());
                             }
-                            PersistWorkerCmd::Append(updates, response) => {
-                                for (id, update, upper) in updates {
-                                    let (old_span, updates, old_upper) =
-                                        all_updates.entry(id).or_insert_with(|| {
-                                            (
-                                                span.clone(),
-                                                Vec::default(),
-                                                Antichain::from_elem(T::minimum()),
-                                            )
-                                        });
-
-                                    if old_span.id() != span.id() {
-                                        // Link in any spans for `Append`
-                                        // operations that we lump together by
-                                        // doing this. This is not ideal,
-                                        // because we only have a true tracing
-                                        // history for the "first" span that we
-                                        // process, but it's better than
-                                        // nothing.
-                                        old_span.follows_from(span.id());
-                                    }
-                                    updates.extend(update);
-                                    old_upper.join_assign(&Antichain::from_elem(upper));
+                            use futures::StreamExt;
+                            futs.collect::<Vec<_>>().await;
+                        },
+                        cmd = rx.recv() => {
+                            if let Some(cmd) = cmd {
```

Comment:
nit: we could do sth like this to reduce the level of indentation
```
let cmd = if let Some(cmd) = cmd {
    cmd
} else {
    shutdown = true;
    break;
}
```

but also works as is 👌

Reply:
I find this harder to read, so I'll leave it for a future re-write (and to avoid the 3x test retries this one needed).